editor/editor_log.cpp

```diff
@@ -94,15 +113,70 @@ void EditorLog::clear() {
 	_clear_request();
 }
 
-void EditorLog::copy() {
-	_copy_request();
+void EditorLog::add_message(const String &p_msg, MessageType p_type) {
+	if (messages.size() > 0 && messages[messages.size() - 1].text == p_msg) {
+		// If previous message is the same as the new one, increase previous count rather than adding another
+		// instance to the messages list.
+		LogMessage &previous = messages.write[messages.size() - 1];
+		previous.count++;
```

Comment:
Errors can often get repeated but not contiguous, e.g.:
```
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_right'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_left'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_down'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_up'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_right'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_left'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_down'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_up'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_right'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_left'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_down'.
   At: main/input_default.cpp:142.
ERROR: get_action_strength: Request for nonexistent InputMap action 'look_up'.
   At: main/input_default.cpp:142.
```

Typically happens when triggering multiple errors in a given frame which are then repeated each frame. It can also happen with chained errors when the engine prints several error messages for a single fault (e.g. when a scene can't be loaded due to missing imported dependencies).

Is this something we could handle? That would require storing all messages in a hashmap to increment the counter regardless of the position, that might impact performance I guess.

Alternatively this could maybe be done over e.g. the last 10 messages, so it would group related duplicate messages which happen in close sequence, but not a later similar message spam from another source. (E.g. physics errors triggered by two different physics body that the player interacts with at different times.)

Reply:
We could look into this I guess but I am hesitant for now, as displaying them "grouped" may mess with the order in which they occur.